app/controllers/authenticate_controller.rb

```diff
@@ -181,6 +181,9 @@ def handle_authentication_error(err)
       Errors::Authentication::AuthnK8s::CertMissingCNEntry
       raise ArgumentError
 
+    when Errors::Authentication::IllegalStateMissingConstraints
+      raise InternalServerError
```

Comment:
why are you returning 500 in this case? is there a reason we do not return 401 like we do with other validation failures?

Reply:
See comment above

Reply:
is this still used?

Reply:
No, i will delete it